python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -557,6 +557,11 @@ def asset_key(self) -> AssetKey:
                 "Cannot call `context.asset_key` in a multi_asset with more than one asset. Use"
                 " `context.asset_key_for_output` instead."
             )
+        observable_asset_key = self.job_def.asset_layer.observable_asset_key_for_node(
+            self.node_handle
+        )
+        if observable_asset_key:
+            return observable_asset_key
```

Comment:
would be more comfortable if precedence was given to how things currently work / the common path and this was added as a fallback to make it work when it previously would have errored